task_list/config/routes.rb

```diff
@@ -8,6 +8,10 @@
   get 'tasks/:id' => 'tasks#show'
   post 'tasks/' => 'tasks#create'
   delete 'tasks/:id' => 'tasks#destroy'
+  get 'tasks/:id/edit' => 'tasks#edit', as: 'edit_task'
+  patch 'tasks/:id' => 'tasks#update'
+  patch 'tasks/:id' => 'tasks#toggle_complete'
```

Comment:
I think the route on the line above might be "hiding" this one. AFAIK, Rails will only accept/match a single route for a given combination of an HTTP verb (`patch`) and URL path (`tasks/:id`).

To make sure that the `tasks#toggle_complete` controller action is available, you should change the URL path to something else (like `tasks/:id/complete`).